crates/biome_js_analyze/src/analyzers/nursery/use_shorthand_assign.rs

```diff
@@ -0,0 +1,214 @@
+                JsBinaryExpression::cast_ref(param.expression().ok()?.syntax())?
+            }
+            _ => return None,
+        };
+
+        let operator = binary_expression.operator().ok()?;
+        let shorthand_operator = get_shorthand(operator)?;
+        let is_commutative = is_commutative(operator);
+        let variable_position_in_expression =
+            parse_variable_reference_in_expression(&left_var_name, &binary_expression)?;
+
+        let replacement_expression = match variable_position_in_expression {
+            VariablePosition::Left => binary_expression.left().ok()?,
+            VariablePosition::Right => binary_expression.right().ok()?,
+        };
+
+        if !is_commutative && matches!(variable_position_in_expression, VariablePosition::Right) {
+            return None;
+        }
+
+        Some(RuleState {
+            shorthand_operator,
+            replacement_expression,
+        })
+    }
+
+    fn diagnostic(ctx: &RuleContext<Self>, state: &Self::State) -> Option<RuleDiagnostic> {
+        let node = ctx.query();
+        let shorthand_operator = state.shorthand_operator.to_string()?;
+
+        Some(RuleDiagnostic::new(
+            rule_category!(),
+            node.range(),
+            markup! {
+                "Assignment "<Emphasis>"(=)"</Emphasis>" can be replaced with operator assignment "<Emphasis>""{shorthand_operator}""</Emphasis>""
+            },
+        ))
+    }
+
+    fn action(ctx: &RuleContext<Self>, state: &Self::State) -> Option<JsRuleAction> {
+        let node = ctx.query();
+
+        let mut mutation = ctx.root().begin();
+        let shorthand_operator = state.shorthand_operator;
+
+        let token_leading_trivia = node.operator_token().ok()?.leading_trivia().pieces();
+        let token_trailing_trivia = node.operator_token().ok()?.trailing_trivia().pieces();
+        let token = make::token(shorthand_operator)
+            .with_leading_trivia_pieces(token_leading_trivia)
+            .with_trailing_trivia_pieces(token_trailing_trivia);
+
+        let shorthand_node = node
+            .clone()
+            .with_operator_token_token(token)
+            .with_right(state.replacement_expression.clone());
+
+        mutation.replace_node(
+            AnyJsExpression::JsAssignmentExpression(node.clone()),
+            AnyJsExpression::JsAssignmentExpression(shorthand_node),
+        );
+
+        Some(JsRuleAction {
+            category: ActionCategory::QuickFix,
+            applicability: Applicability::MaybeIncorrect,
+            message: markup! { "Use "<Emphasis>""{shorthand_operator.to_string()?}""</Emphasis>" instead." }
+                .to_owned(),
+            mutation,
+        })
+    }
+}
+
+fn parse_variable_reference_in_expression(
+    variable_name: &str,
+    binary_expression: &JsBinaryExpression,
+) -> Option<VariablePosition> {
+    let present_on_left = variable_name == binary_expression.left().ok()?.omit_parentheses().text();
```

Comment:
We could avoid unnecessary work (checking right) by returning early.
```suggestion
    let present_on_left = variable_name == binary_expression.left().ok()?.omit_parentheses().text();
    if present_on_left {
        return Some(VariablePosition::Left);
    }
```